### ISSUE TITLE
action-alias update : 500 Internal Server Error

### ISSUE THREAD

**Issue by humblearner** [StackStorm/st2#3170]
`$ st2 --debug action-alias update examples.remote_shell_cmd remote_shell_cmd.yaml`
```
# -------- begin 140245180802000 request ----------
curl -X GET -H  'Connection: keep-alive' -H  'Accept-Encoding: gzip, deflate' -H  'Accept: */*' -H  'User-Agent: python-requests/2.11.1' -H  'X-Auth-Token: 45baab7d9d4b4124aed3fb6781c9e9dd' 'http://127.0.0.1:9101/v1/actionalias/?name=examples.remote_shell_cmd'
# -------- begin 140245180802000 response ----------
[]
# -------- end 140245180802000 response ------------

# -------- begin 140245180802000 request ----------
curl -X GET -H  'Connection: keep-alive' -H  'Accept-Encoding: gzip, deflate' -H  'Accept: */*' -H  'User-Agent: python-requests/2.11.1' -H  'X-Auth-Token: 45baab7d9d4b4124aed3fb6781c9e9dd' http://127.0.0.1:9101/v1/actionalias/examples.remote_shell_cmd
# -------- begin 140245180802000 response ----------
{"description": "Execute a command on a remote host via SSH.", "extra": {}, "ack": {"format": "Hold tight while I run command: *{{execution.parameters.cmd}}* on hosts *{{execution.parameters.hosts}}*"}, "enabled": true, "name": "remote_shell_cmd", "result": {"format": "Ran command *{{execution.parameters.cmd}}* on *{{ execution.result | length }}* hosts.\n\nDetails are as follows:\n{% for host in execution.result -%}\n    Host: *{{host}}*\n    ---> stdout: {{execution.result[host].stdout}}\n    ---> stderr: {{execution.result[host].stderr}}\n{%+ endfor %}\n"}, "formats": ["run {{cmd}} on {{hosts}}"], "action_ref": "core.remote", "pack": "examples", "ref": "examples.remote_shell_cmd", "id": "588a97e255fc8c34bd710e02", "uid": "action:examples:remote_shell_cmd"}
# -------- end 140245180802000 response ------------

# -------- begin 140245180802000 request ----------
curl -X PUT -H  'Connection: keep-alive' -H  'Accept-Encoding: gzip, deflate' -H  'Accept: */*' -H  'User-Agent: python-requests/2.11.1' -H  'content-type: application/json' -H  'X-Auth-Token: 45baab7d9d4b4124aed3fb6781c9e9dd' -H  'Content-Length: 660' --data-binary '{"description": "Execute a command on a remote host via SSH.", "ack": {"format": "Hold tight while I run command: *{{execution.parameters.cmd}}* on hosts *{{execution.parameters.hosts}}*"}, "name": "remote_shell_cmd", "result": {"format": "Ran command *{{execution.parameters.cmd}}* on *{{ execution.result | length }}* hosts.\n\nDetails are as follows:\n{% for host in execution.result -%}\n    Host: *{{host}}*\n    ---> stdout: {{execution.result[host].stdout}}\n    ---> stderr: {{execution.result[host].stderr}}\n{%+ endfor %}\n"}, "formats": ["run {{cmd}} on {{hosts}}"], "action_ref": "core.remote", "id": "588a97e255fc8c34bd710e02", "pack": "examples"}' http://127.0.0.1:9101/v1/actionalias/588a97e255fc8c34bd710e02
# -------- begin 140245180802000 response ----------
{
    "faultstring": "Internal Server Error"
}
# -------- end 140245180802000 response ------------

ERROR: 500 Server Error: Internal Server Error
MESSAGE: Internal Server Error for url: http://127.0.0.1:9101/v1/actionalias/588a97e255fc8c34bd710e02

CLI settings:
----------------
Config file path: /home/vagrant/.st2/config
Client settings:
----------------
ST2_BASE_URL: http://127.0.0.1
ST2_AUTH_URL: http://127.0.0.1:9100
ST2_API_URL: http://127.0.0.1:9101/v1
ST2_AUTH_TOKEN: 45baab7d9d4b4124aed3fb6781c9e9dd

Proxy settings:
---------------
HTTP_PROXY:
HTTPS_PROXY:

Traceback (most recent call last):
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2client/shell.py", line 278, in run
    args.func(args)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2client/commands/resource.py", line 418, in run_and_print
    instance = self.run(args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2client/commands/resource.py", line 44, in decorate
    return func(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2client/commands/resource.py", line 415, in run
    return self.manager.update(modified_instance, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2client/models/core.py", line 39, in decorate
    return func(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2client/models/core.py", line 304, in update
    self.handle_error(response)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2client/models/core.py", line 165, in handle_error
    response.raise_for_status()
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/requests/models.py", line 862, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 500 Server Error: Internal Server Error
MESSAGE: Internal Server Error for url: http://127.0.0.1:9101/v1/actionalias/588a97e255fc8c34bd710e02
```

st2api.log:

```
2017-01-27 01:18:36,872 139883466543664 INFO hooks [-] 28899533-4eb4-4709-9326-3c166f598059 -  GET /v1/actionalias/ with filters={u'name': u'examples.remote_shell_cmd'} (remote_addr='127.0.0.1',method='GET',filters={u'name': u'examples.remote_shell_cmd'},request_id='28899533-4eb4-4709-9326-3c166f598059',path='/v1/actionalias/')
2017-01-27 01:18:36,872 139883466543664 AUDIT auth [-] Token provided in headers
2017-01-27 01:18:36,874 139883466543664 AUDIT auth [-] Token with id "588a821b55fc8c5d2c01592f" is validated.
2017-01-27 01:18:36,876 139883466543664 INFO resource [-] GET all /v1/actionalias/ with filters={'order_by': ['pack', 'name'], 'name': u'examples.remote_shell_cmd'} (offset=0,limit='None',filters={'order_by': ['pack', 'name'], 'name': u'examples.remote_shell_cmd'},sort=[])
2017-01-27 01:18:36,878 139883466543664 INFO hooks [-] 28899533-4eb4-4709-9326-3c166f598059 - GET /v1/actionalias/ (remote_addr='127.0.0.1',method='GET',status_code='200 OK',request_id='28899533-4eb4-4709-9326-3c166f598059',path='/v1/actionalias/')
2017-01-27 01:18:36,888 139883466543664 INFO hooks [-] 17f01cbc-36fd-4497-8768-9f7654368c1e -  GET /v1/actionalias/examples.remote_shell_cmd with filters={} (remote_addr='127.0.0.1',method='GET',filters={},request_id='17f01cbc-36fd-4497-8768-9f7654368c1e',path='/v1/actionalias/examples.remote_shell_cmd')
2017-01-27 01:18:36,889 139883466543664 AUDIT auth [-] Token provided in headers
2017-01-27 01:18:36,890 139883466543664 AUDIT auth [-] Token with id "588a821b55fc8c5d2c01592f" is validated.
2017-01-27 01:18:36,894 139883466543664 INFO resource [-] GET /v1/actionalias/examples.remote_shell_cmd with ref_or_id=examples.remote_shell_cmd
2017-01-27 01:18:36,897 139883466543664 INFO hooks [-] 17f01cbc-36fd-4497-8768-9f7654368c1e - GET /v1/actionalias/examples.remote_shell_cmd result={"description": "Execute a command on a remote host via SSH.", "extra": {}, "ack": {"format": "Hold tight while I run command: *{{execution.parameters.cmd}}* on hosts *{{execution.parameters.hosts}}*"}, "enabled": true, "name": "remote_shell_cmd", "result": {"format": "Ran command *{{execution.parameters.cmd}}* on *{{ execution.result | length }}* hosts.\n\nDetails are as follows:\n{% for host in execution.result -%}\n    Host: *{{host}}*\n    ---> stdout: {{execution.result[host].stdout}}\n    ---> stderr: {{execution.result[host].stderr}}\n{%+ endfor %}\n"}, "formats": ["run {{cmd}} on {{hosts}}"], "action_ref": "core.remote", "pack": "examples", "ref": "examples.remote_shell_cmd", "id": "588a97e255fc8c34bd710e02", "uid": "action:examples:remote_shell_cmd"} (result='{"description": "Execute a command on a remote host via SSH.", "extra": {}, "ack": {"format": "Hold tight while I run command: *{{execution.parameters.cmd}}* on hosts *{{execution.parameters.hosts}}*"}, "enabled": true, "name": "remote_shell_cmd", "result": {"format": "Ran command *{{execution.parameters.cmd}}* on *{{ execution.result | length }}* hosts.\\n\\nDetails are as follows:\\n{% for host in execution.result -%}\\n    Host: *{{host}}*\\n    ---> stdout: {{execution.result[host].stdout}}\\n    ---> stderr: {{execution.result[host].stderr}}\\n{%+ endfor %}\\n"}, "formats": ["run {{cmd}} on {{hosts}}"], "action_ref": "core.remote", "pack": "examples", "ref": "examples.remote_shell_cmd", "id": "588a97e255fc8c34bd710e02", "uid": "action:examples:remote_shell_cmd"}',request_id='17f01cbc-36fd-4497-8768-9f7654368c1e',status_code='200 OK',remote_addr='127.0.0.1',path='/v1/actionalias/examples.remote_shell_cmd',method='GET')
2017-01-27 01:18:36,904 139883466543664 INFO hooks [-] 381bffb3-c6b3-44f5-84ca-a7e896935301 -  PUT /v1/actionalias/588a97e255fc8c34bd710e02 with filters={} (remote_addr='127.0.0.1',method='PUT',filters={},request_id='381bffb3-c6b3-44f5-84ca-a7e896935301',path='/v1/actionalias/588a97e255fc8c34bd710e02')
2017-01-27 01:18:36,905 139883466543664 AUDIT auth [-] Token provided in headers
2017-01-27 01:18:36,907 139883466543664 AUDIT auth [-] Token with id "588a821b55fc8c5d2c01592f" is validated.
2017-01-27 01:18:36,916 139883466543664 WARNING base [-] Type definition for 'action_alias' argument of 'put' is missing.
2017-01-27 01:18:36,916 139883466543664 ERROR hooks [-] API call failed: argument of type 'ActionAliasAPI' is not iterable
Traceback (most recent call last):
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pecan/core.py", line 631, in __call__
    self.invoke_controller(controller, args, kwargs, state)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pecan/core.py", line 531, in invoke_controller
    result = controller(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/rbac/decorators.py", line 122, in func_wrapper
    return func(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/models/api/base.py", line 281, in callfunction
    raise e
TypeError: argument of type 'ActionAliasAPI' is not iterable (_exception_data={},_exception_class='TypeError',_exception_message="argument of type 'ActionAliasAPI' is not iterable")
2017-01-27 01:18:36,918 139883466543664 ERROR hooks [-] Traceback (most recent call last):
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pecan/core.py", line 631, in __call__
    self.invoke_controller(controller, args, kwargs, state)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pecan/core.py", line 531, in invoke_controller
    result = controller(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/rbac/decorators.py", line 122, in func_wrapper
    return func(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/models/api/base.py", line 281, in callfunction
    raise e
TypeError: argument of type 'ActionAliasAPI' is not iterable
Traceback (most recent call last):
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pecan/core.py", line 631, in __call__
    self.invoke_controller(controller, args, kwargs, state)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/pecan/core.py", line 531, in invoke_controller
    result = controller(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/rbac/decorators.py", line 122, in func_wrapper
    return func(*args, **kwargs)
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/models/api/base.py", line 281, in callfunction
    raise e
TypeError: argument of type 'ActionAliasAPI' is not iterable
2017-01-27 01:18:36,921 139883466543664 INFO hooks [-] 381bffb3-c6b3-44f5-84ca-a7e896935301 - PUT /v1/actionalias/588a97e255fc8c34bd710e02 result={
    "faultstring": "Internal Server Error"
} (result='{\n    "faultstring": "Internal Server Error"\n}',request_id='381bffb3-c6b3-44f5-84ca-a7e896935301',status_code='500 Internal Server Error',remote_addr='127.0.0.1',path='/v1/actionalias/588a97e255fc8c34bd710e02',method='PUT')
```

**Comment by Kami** [comment]
Good catch, will look into a fix.

**Comment by lakshmi-kannan** [comment]
Seems fixed. Closing. cc: @Kami

**Comment by Kami** [comment]
Yep, fixed in #3184. Thanks for closing.
